### PR TITLE
xdr: update using stellar/xdrgen#68

### PR DESCRIPTION
**Pull request by 2opremio**
It improves XDR marshaling performance.

See stellar/xdrgen#68 for details

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -40,7 +40,6 @@ namespace :xdr do
     require "pathname"
     require "xdrgen"
     require 'fileutils'
-    FileUtils.rm_f("xdr/xdr_generated.go")
```
Comment (leighmcculloch) [inline Rakefile:43]: ❗ This looks unrelated and like an unnecessary change. Why is the change needed? I believe this is there to make sure that any xdr_generated.go we have at the end of a generate run is new and no silent error that doesn't result in a new file being written goes undetected.

File: Gemfile.lock [hunk 2]
```diff
@@ -41,7 +41,7 @@ GEM
     faraday-net_http_persistent (1.2.0)
     faraday-patron (1.0.0)
     faraday-rack (1.0.0)
-    i18n (1.8.10)
+    i18n (1.8.11)
```
Comment (leighmcculloch) [inline Gemfile.lock:44]: ❗ It looks like you probably ran `bundle update xdrgen` and it has updated another dependency that is unnecessary. Ruby's bundler will update any dependency it can. Can you change this back to 1.8.10? I think we can avoid updating other dependencies unless we see a need.

**Review by leighmcculloch — APPROVED** [review]
Two requests (❗), otherwise looks great. 🎉

**Review by bartekn — APPROVED** [review]
LGTM.

**Comment by bartekn** [comment]
I was curious and run CPU profiler on Horizon 2.10.0 and master. Highlighted all `xdr.*` calls. `TransactionProcessor` is now faster then `OperationProcessor` on my machine! Great job @leighmcculloch and @2opremio!

`horizon-v2.10.0`
Screenshot 2021-11-10 at 23 46 11

`master`
Screenshot 2021-11-10 at 23 46 21

File: Rakefile (hunk 1 above)
Comment (2opremio) [reply Rakefile:43]: It made development unpleasant because my IDE closed the buffer when the file was removed.

There is lo compelling reason to remove the file, since it's overwritten.

File: Gemfile.lock (hunk 2 above)
Comment (2opremio) [reply Gemfile.lock:44]: I run a conservative update, only on xdrgen.

Comment (leighmcculloch) [reply Gemfile.lock:44]: Even on conservative updates bundler will update all transitive dependencies for the gem you requested be updated.

File: Rakefile (hunk 1 above)
Comment (leighmcculloch) [reply Rakefile:43]: Ah interesting. We can solve both problems by replacing the `rm` with emptying the file. So the file never disappears, but it still behaves as before and is zeroed out?

Comment (2opremio) [reply Rakefile:43]: I still don't see a compelling reason to reset the file.

File: Gemfile.lock (hunk 2 above)
Comment (2opremio) [reply Gemfile.lock:44]: I don't see it as a very important side effect.

**Comment by 2opremio** [comment]
@bartekn is there a benchmark for that? If so, we should run it when optimizing Horizon. If not, I think we should create one.

**Comment by 2opremio** [comment]
I think we could have a benchmark for each transaction processor and one for ingestion in general.

**Comment by bartekn** [comment]
I think benchmark makes sense but only with mocked DB to filter it out. I think we can do (in a separate PR) but we are constantly monitoring Horizon metrics so not sure if it's required.

**Comment by 2opremio** [comment]
It would be really helpful when working on profiling (not just for monitor how Horizon's performance is going). I think it would be useful even when running against a local DB.
